codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustWriter.kt

```diff
@@ -439,16 +440,56 @@ fun RustWriter.deprecatedShape(shape: Shape): RustWriter {
 fun <T : AbstractCodeWriter<T>> T.escape(text: String): String =
     text.replace("$expressionStart", "$expressionStart$expressionStart")
 
-/** Parse input as HTML and normalize it */
+/** Parse input as HTML and normalize it, for suitable use within Rust docs. */
 fun normalizeHtml(input: String): String {
     val doc = Jsoup.parse(input)
     doc.body().apply {
-        normalizeAnchors() // Convert anchor tags missing href attribute into pre tags
+        normalizeAnchors()
+        escapeBracketsInText()
     }
 
     return doc.body().html()
 }
 
+/**
+ * Escape brackets in elements' inner text.
+ *
+ * For example:
+ *
+ * ```html
+ * <body>
+ *     <p>Text without brackets</p>
+ *     <div>Some text with [brackets]</div>
+ *     <span>Another [example, 1]</span>
+ * </body>
+ * ```
+ *
+ * gets converted to:
+ *
+ * ```html
+ * <body>
+ *     <p>Text without brackets</p>
+ *     <div>Some text with \[brackets\]</div>
+ *     <span>Another \[example, 1\]</span>
+ * </body>
+ * ```
+ *
+ * This is useful when rendering Rust docs, since text within brackets might get misinterpreted as broken Markdown doc
+ * links (`[link text](https://example.com)`).
+ **/
+private fun Element.escapeBracketsInText() {
```

Comment:
Nit: Just a thought: Should we do this *only if* the box brackets are not followed by text in parentheses? This would allow Smithy model authors to write linkable text in their documentation. For example, `[this is clickable](https://example.com)` should not be escaped.